src/tools/pr_8799/src/main.rs

```diff
@@ -0,0 +1,55 @@
+mod virtcontainers;
+
+use anyhow::{anyhow, Result};
+use std::collections::HashMap;
+use std::process::{exit, Command};
+use virtcontainers::get_ch_vcpu_tids_by_path;
+
+fn get_ch_vcpu_tids(pid: u32) -> Result<HashMap<u32, u32>> {
+    let proc_path = format!("/proc/{pid}");
+
+    let map = match get_ch_vcpu_tids_by_path(&proc_path) {
+        Ok(i) => i,
+        Err(e) => {
+            return Err(anyhow!("{:?}", e));
+        }
+    };
```

Comment:
I think you can simplify this:

```suggestion
    let map = get_ch_vcpu_tids_by_path(&proc_path)?;
```